### ISSUE TITLE
False positive 'not-callable' for `namedtuple` attribute of `Enum`

### ISSUE THREAD

**Issue by allanlewis** [PyCQA/pylint#1919]
### Steps to reproduce
Create a subclass of `enum.Enum` (I'm using Python 2.7) with a `collections.namedtuple` attribute and create an instance of the `namedtuple` as another attribute:
```py
class Foo(Enum):
    _Bar = namedtuple('Bar', 'x y')
    bar = _Bar(1, 2)
```

### Current behavior
Pylint reports `not-callable` against the `bar = Bar...` line.

### Expected behavior
Pylint should not report `not-callable` for this code.

### pylint --version output
```
pylint 1.6.5, 
astroid 1.4.9
Python 2.7.11 (default, Sep 29 2016, 13:33:00) 
[GCC 5.3.1 20160406 (Red Hat 5.3.1-6)]
```

Perhaps Pylint special-cases `Enum`s and treats `_Bar` as a member. It's arguably a bad use of `Enum`, but `not-callable` is nonetheless incorrect.

**Comment by PCManticore** [comment]
Thanks for the report! pylint 1.6.5 is a pretty old version, this issue seems to be solved with the latest 1.8 release.

**Comment by meuter** [comment]
Sorry to reopen this issue, but I'm currently running pylint 2.8.2 on windows and I still get the false positive.

```
(env) PS C:\temp> pylint --version
pylint 2.8.2
astroid 2.5.6
Python 3.9.4 (tags/v3.9.4:1f2e308, Apr  6 2021, 13:40:21) [MSC v.1928 64 bit (AMD64)]
(env) PS C:\temp> type bug.py     
from typing import NamedTuple

Bar = NamedTuple("Bar", foo=int, baz=str)     
Bar(10, "yo")
(env) PS C:\temp> pylint bug.py          
************* Module bug
bug.py:4:0: E1102: Bar is not callable (not-callable)

--------------------------------------------------------------------
Your code has been rated at -6.67/10 (previous run: -6.67/10, +0.00)
```

Am I missing something?

**Comment by Pierre-Sassoulas** [comment]
There is no problem with ``collections.namedtuple`` but there is with ``typing.NamedTuple``. Could you open a new issue please @meuter ?

**Comment by meuter** [comment]
Sure, no problem 👍
